zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -578,6 +658,21 @@ private int getRiskId(JSONObject parameters) throws ApiException {
         return riskId;
     }
 
+    private RecordHistory getRecordHistory(TableHistory tableHistory, Integer id)
+            throws ApiException {
+        RecordHistory recordHistory;
+        try {
+            recordHistory = tableHistory.read(id);
+        } catch (HttpMalformedHeaderException | DatabaseException e) {
+            logger.error("Failed to read the history record:", e);
+            throw new ApiException(ApiException.Type.INTERNAL_ERROR, e);
+        }
+        if (recordHistory == null) {
+            throw new ApiException(ApiException.Type.DOES_NOT_EXIST, Integer.toString(id));
+        }
+        return recordHistory;
+    }
+
```

Comment:
I ripped this code from CoreAPI (instead of making that method `public static`, I'm happy to do it another way if there are suggestions. (Or we could move it to some sort of APIUtils class?)

Reply:
Yeah, `ApiUtils` would be a good place (if really needed, per comment about the `HistoryReference`).